controllers/Handle.php

```diff
@@ -48,72 +52,116 @@ public function __construct()
         $this->authManager = BackendAuth::instance();
 
         $this->enabledProviders = Config::get('winter.sso::enabled_providers', []);
+
+        Event::listen('backend.auth.login', function ($user) {
+            $runMigrationsOnLogin = (bool) Config::get('cms.runMigrationsOnLogin', Config::get('app.debug', false));
+            if ($runMigrationsOnLogin) {
+                try {
+                    // Load version updates
+                    UpdateManager::instance()->update();
+                } catch (Exception $e) {
+                    Flash::error($e->getMessage());
+                }
+            }
+            // Log the sign in event
+            AccessLog::add($user);
+        });
```

Comment:
Why is this being added in this plugin instead of in the backend module?

Reply:
it was in here already , I just moved it.

Reply:
Fixed. Also see https://github.com/wintercms/winter/pull/1257